tests/test_needs_external_needs_build.py

```diff
@@ -36,7 +36,10 @@ def test_doc_build_html(test_app, sphinx_test_tempdir):
         ["sphinx-build", "-b", "html", "-D", rf"plantuml={plantuml}", src_dir, out_dir],
         capture_output=True,
     )
-    assert not output_second.stderr
+    assert output_second.stderr.decode("utf-8").splitlines() == [
```

Comment:
it seems that previously this was incorrect; one would assume if you get these warnings on the first build, then there is no reason you should not get them on a cached re-build